comp/core/agenttelemetry/impl/sender.go

```diff
@@ -166,23 +166,18 @@ func newSenderImpl(
 	logComp log.Component,
 	client client) (sender, error) {
 
-	// Form Agent Telemetry endpoint URL
-	// We cannot use utils.GetMainEndpoint() function call with "dd_url" parameters directly,
-	// at this point, just as other Agent components are used for forwarding data to Datadog
-	// for a number of reasons. Accordingly, "site" configuration is required to be used at
-	// this time. This will fail for an explicit proxy relying on dd_url configuration. In
-	// future it will be addressed either at intake or at the Agent level (APM Telemetry e.g.
-	// is required to use "apm_config.telemetry.dd_url").
-	//
-	// On related note "sending" part of the sender will be moved to "DefaultForwarder"
+	// "Sending" part of the sender will be moved to EP Forwarder in the future to be able
 	// to support retry, caching, URL management, API key rotation at runtime, flush to
-	// disk, backoff logic, etc.
-	site := cfgComp.GetString("site")
-	if len(site) == 0 {
-		return nil, fmt.Errorf("site is not set in the configuration")
+	// disk, backoff logic, etc. Specifically, different types of data need to be sent using
+	// different types of EP Forwarder Pipelines (separate entries for passthroughPipelineDesc
+	// array).
+
+	// build endpoint URL
+	endpointHost := utils.GetMainEndpoint(cfgComp, telemetryEndpointPrefix, "agent_telemetry.dd_url")
+	if len(endpointHost) == 0 {
```

Comment:
I may be missing something, but I don't think `GetMainEndpoint` can ever return an empty string. Other users also do not expect an empty response, so I think this check can be removed.

Reply:
Agree, it become redundant, thank you. The fix is in the latest commit